Update folder walkthrough [DOC-5]

johnnonweiler opened the pull request:
Remove links to folders that no longer exist (clobber and memory-models), and add directory dependencies diagram.

Commit:
Remove links to clobber and memory-models in doc

Commit:
Add directory dependencies diagram

chrisr-diffblue commented on doc/architectural/folder-walkthrough.md:
```diff
@@ -84,3 +82,110 @@ develop regression tests.
+  CBMC -> { goto_symex, linking, ansi_c, cpp, jsil };
+  goto_cc -> { linking, ansi_c, cpp, jsil };
+  goto_instrument -> pointer_analysis;
+  goto_analyzer -> pointer_analysis;
```
Don't goto_instrument/goto_analyzer also depend on `analysis` directly? (and not just transitively through `pointer_analysis`)

johnnonweiler replied:
@peterschrammel ?

NlightNFotis replied:
@chrisr-diffblue Yes, at least `goto-instrument/aggressive_slicer` depends on `analyses`

peterschrammel replied:
I think the arrows should be read transitively, otherwise the diagram becomes unreadable.

NlightNFotis reviewed (approved):
Looks good to me. Pending CI managing to build it without issues this is a green light from me.

tautschnig commented on doc/architectural/folder-walkthrough.md:
```diff
@@ -82,3 +82,110 @@ develop regression tests.
+    CBMC -> goto_instrument;
+    janalyzer -> goto_analyzer;
+    jdiff -> goto_diff;
+    JBMC -> CBMC;
```
There's also smt2_solver

tautschnig commented on doc/architectural/folder-walkthrough.md:
```diff
@@ -82,3 +82,110 @@ develop regression tests.
+    util -> big_int;
+    util -> nonstd;
+    util -> json;
+    util -> xmllang;
```
Is that actually the case?

peterschrammel replied:
No, it should be json -> util, xmllang -> util and then goto-programs uses json and xmllang.

cesaro commented:
For reference, on the left-hand side we see the graph included in this PR; on the right-hand side we see the one made by @peterschrammel. Since the nodes in the one of the right have been manually located I find it easier to read:

![image](https://user-images.githubusercontent.com/143977/44517866-ab241380-a6c0-11e8-9768-bf35448ee550.png)

Isn't it better to slightly improve the one on the right and then include a screenshot? Or build the diagram in Inkscape and commit an editable .svg file, and then put a screenshot here?

johnnonweiler commented:
I have now replaced the diagram which was based on the one drawn by @peterschrammel with a new diagram based on the dependencies listed in the `module_dependencies.txt` files (and some minor modifications based on conversation with @peterschrammel).  I have also added a couple of sentences of explanation above the diagram.

The new diagram includes the changes suggested by @tautschnig, but also includes other differences.  (Since transitively included dependencies are not drawn directly, the new diagram looks significantly different.)

This is what it looks like now:
![dot_inline_dotgraph_2](https://user-images.githubusercontent.com/32836213/44585810-6d4ceb00-a7a5-11e8-8704-e74717123ac2.png)

@cesaro I agree that the manually drawn diagram was easier to read as you showed above.  However, I think this updated diagram is relatively easy to read now.  Also, using the dot format should make the diagram easier to maintain.

peterschrammel commented on doc/architectural/folder-walkthrough.md:
```diff
@@ -84,3 +82,124 @@ develop regression tests.
+    janalyzer [URL = "\ref janalyzer"];
+    jdiff [URL = "\ref jdiff"];
+    JBMC [URL = "\ref jbmc"];
+    smt2;
```
`smt2_solver`

peterschrammel commented:
I also think it's easy to read. And it is definitely much easier to maintain.
